src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -430,7 +435,12 @@ private ServiceContext(
       this.cleanerFileNames = new LinkedList<>();
       this.buffer = new SnowpipeBuffer();
       this.ingestionService = conn.buildIngestService(stageName, pipeName);
-      this.prefix = FileNameUtils.filePrefix(conn.getConnectorName(), tableName, partition);
+      // SNOW-1642799 = if multiple topics load data into single table, we need to ensure prefix is unique per table - otherwise, file cleaners for different channels may run into race condition
+      if (determineTopic2TableMode(topic2TableMap, topicName) == TopicToTableModeExtractor.Topic2TableMode.MANY_TOPICS_SINGLE_TABLE) {
+        this.prefix = FileNameUtils.filePrefix(conn.getConnectorName(), tableName, topicName, partition);
+      } else {
+        this.prefix = FileNameUtils.filePrefix(conn.getConnectorName(), tableName, "", partition);
```

Comment:
we could enable this for all connectors, but this would mean breaking backward compatibility - for scenarios, where there is just 1:1 mapping between topic and table - this fix doesn't change logic, but the changed prefix would mean there could be garbage files left on stage after upgrading/downgrading. 
for scenarios when there is  n:1 mapping - there would be garbage on stage, requiring manual cleanup, but customer would get the fixed connector instead.

Reply:
We're breaking backward compatibility for some of the customers anyway. Isn't it simpler to make a breaking change for everyone and simplify the filename generation logic?

Reply:
taking into account, that only 3 customers reported this specific error after migration (based on the spreadsheet Xin provided a few days ago, we've got over 200 customers using KC with affected version - (for simplicity >= 2.3.0) using SNOWPIPE ingestion, I think it would be better to keep the generation backwards compatible for them. It seems the configuration, where multiple topics feed single table is rather exception than the rule - otherwise we would probably see much more raised incidents.